metadata/metadata.go

```diff
@@ -315,6 +317,11 @@ func (lookup localResourceLookup) HasJob(id ResourceID) (bool, error) {
 	return false, nil
 }
 
+type NameVariantKey struct {
+	Name    string
+	Variant string
+}
+
```

Comment:
@simba-git, let me know if you'd like to change the naming here.

Reply:
Would make this lowercased so its not externalized outside the package (in Go, lowercase = private). You could also put this struct in the method where it's used to further limit the scope